lib/livebook/file_systems.ex

```diff
@@ -61,4 +61,17 @@ defmodule Livebook.FileSystems do
   def module_to_type(module)
   def module_to_type(FileSystem.Local), do: "local"
   def module_to_type(FileSystem.S3), do: "s3"
+
+  @doc """
+  Returns default directory based on given hub.
+  """
+  @spec default_file(Livebook.Hubs.Provider.t()) :: FileSystem.File.t()
+  def default_file(hub) do
+    file_systems = Livebook.Hubs.get_file_systems(hub)
+    file = Livebook.Settings.default_dir()
```

Comment:
Let's call this function `default_dir(hub)`, so it's a 1-arity version of `default_dir/0`.